e2e/libs/keywords/recurringjob_keywords.py

```diff
@@ -0,0 +1,51 @@
+from recurringjob import RecurringJob
+
+from utility.resource_cache import ResourceCache
+from utility.utility import logging
+
+from utility.constant import KIND_RECURRINGJOB
+from utility.constant import KIND_VOLUME
+
+
+class recurringjob_keywords:
+
+    def __init__(self):
+        self.recurringjob = RecurringJob()
+
+    def cleanup_recurringjobs(self):
+        cached_recurringjobs = ResourceCache.get_kind(KIND_RECURRINGJOB)
+        if cached_recurringjobs is None:
+            return
+
+        recurringjob_names = cached_recurringjobs.keys()
+        self.recurringjob.cleanup(recurringjob_names)
+        ResourceCache.reset_kind(KIND_RECURRINGJOB)
+
+    def create_snapshot_recurringjob_for_volume(self, volume_name):
+        job_name = volume_name + '-snap'
+        self.recurringjob.create(job_name, task="snapshot")
+        ResourceCache.new(KIND_RECURRINGJOB, job_name)
+
+        self.recurringjob.add_to_volume(job_name, volume_name)
+        ResourceCache.create_or_update_link(
+            KIND_VOLUME, volume_name, KIND_RECURRINGJOB, job_name
+        )
```

Comment:
Don't need to have cache. Let's query it at runtime if need to know the jobs belonging to the volume. Kubernetes way.